### DISCUSSION TITLE
Deepgram disconnecting

### DISCUSSION THREAD

**Discussion by eeejayg** [deepgram/community#913] [General help] [answered]
I am trying to get the starter project going for Node.js / Twilio. I have a Twilio stream coming in that I am trying to transcribe.

Regardless of which socket server I use (uWebsockets, ws, Faye, I've tried 'em all) my Deepgram keeps disconnecting. From multiple different socket servers I am getting this metadata message:

uWS: client connected
deepgram: CONNECTED
deepgram: metadata received
data:  {"metadata":{"type":"Metadata","transaction_key":"deprecated","request_id":"5af15841-8596-418a-933b-93c434ac80fb","sha256":"5ffb9af2ac805ab1ce0b2b808de369d7faf16fd77e1ada116ed4ed197362b4b8","created":"2024-09-05T13:57:10.769Z","duration":0,"channels":0}}
deepgram: disconnected

Any help would be greatly appreciated.

My code is based entirely on "https://github.com/deepgram-starters/live-node-starter" server.js code

Running NGINX. No access or error log entries occur. Headers and upgrade are properly set up.

The data flow looks like:

web socket connects
deepgram  connects
wait for a few ms until deepgram readystate =1 
a few Twilio audio messages in the correct format get sent to deepgram
deepgram has a metadata event (see payload above)
deepgram goes into readystate = 3 and stays there
audio messages keep coming in

Thank you.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hi @eeejayg, our logs show you are sending JSON data over the websocket rather than binary audio data. Can you double check what object is being sent?

**Reply by eeejayg** [reply]
Jason:

I wonder if I could prevail upon you to again look at the logs. I've got everything mostly working. I checked the audio by writing it to a file, converting it MP3 and listening to it. So the audio part is working. But I'm not getting any Transcription events. Eventually I get a Metadata event and it poops out.

My Node.js code is attached.

Thank you.

Eric Juvet

[dg-uws.txt](https://github.com/user-attachments/files/16902457/dg-uws.txt)

**Reply by eeejayg** [reply]
I have attached the package.json that goes with the code I sent in the event you care to test it and see what is going on. The phone number associated with this app is 206 944 5891. Call it and it will send a stream to this app.

[package.txt](https://github.com/user-attachments/files/16903047/package.txt)

**Comment by eeejayg** [comment]
Jason:

Wow! Thank you so much for checking the logs. That was really nice of you.

Yes, it does appear to be JSON. It's from a Twilio audio stream. Still
having trouble decoding it. The data property of the message supposedly has
the audio data in Base64.

But I can't tell if it is a string or an array of Char or what. (Seems like
a pretty heavy way to send data, imo).

I'm using Node. Anyone there have experience dealing with Twilio streams?

Many thanks,
Eric Juvet

**Reply by eeejayg** [reply, marked as answer]
Well, I got it figured out. In Node the raw message is seen as Buffer object. You can use the toString method of the Buffer object.

message = Buffer.from(someRawMessageFromSocket).toString();
messageObj = JSON.parse(message);

The Twilio audio stream object looks like:

{
  "event": "media",
  "sequenceNumber": "159",
  "media": {
    "track": "inbound",
    "chunk": "158",
    "timestamp": "3246",
    "payload": "(audio data is a base64 encoded string)"
    },
  "streamSid": "MZccfc1da74da51502c72a046fcad00826"
}

It seems like a pretty heavy way to deliver binary data. Not liking the base64 aspect.
